spirit/core/static/spirit/scripts/src/editor_file_upload.coffee

```diff
@@ -0,0 +1,117 @@
+###
+    Markdown editor image upload, should be loaded before $.editor()
+    requires: util.js
+###
+
+$ = jQuery
+
+
+class EditorFileUpload
+
+    defaults: {
+        csrfToken: "csrf_token",
+        target: "target url",
+        placeholderText: "uploading {file_name}"
+    }
+
+    constructor: (el, options) ->
+        @el = $(el)
+        @options = $.extend({}, @defaults, options)
+        @formFile = $("<form/>")
+        @inputFile = $("<input/>", {
+            type: "file",
+            accept: ".doc, .docx, .pdf"}).appendTo(@formFile)
```

Comment:
This should be configurable. I think we can just pass an array as option. The array would contain `settings.ST_ALLOWED_UPLOAD_FILE_FORMAT`